### PR TITLE
[#3570] change to hs_irods:4.2.6 image

### PR DESCRIPTION
**Pull request by phuongdm**
### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. [Enter positive test case here]

### CODE REVIEW

**Comment by phuongdm** [comment]
@pkdash everything seems OK now, no need to run` chmod /tmp ` anymore 

Screen Shot 2019-09-15 at 9 33 54 AM

**Comment by pkdash** [comment]
@phuongdm Great.

**Comment by phuongdm** [comment]
@sblack-usu , please review on dev-hs-1, hope it does not make trouble like previous PR

**Comment by phuongdm** [comment]
@hyi,

The local-dev settings got this error if run test locally, I don't know why it can pass the Jenkins test (still use iRODS 4.1.8) because even we use rodsadmin user, it cannot upload a file to a personal home dir without chmod. Can you take a look on the test code and make some modify if need? Thanks!

image

**Comment by hyi** [comment]
@phuongdm I just ran that test locally but got ```AssertionError: irods docker containers are not set up properly for federation testing```. I am using the develop branch. Should I pull your PR branch in to test this? The fact that all tests have passed in Jenkins but not in local environment might be a good entry point to figure out the reason for this problem in local dev environment although I don't know the local dev env set up much to provide any useful input here at this point.

**Comment by phuongdm** [comment]
@hyi, 

Yes, you need to pull the branch to test. I copied the iRODS set up from Jenkins to the PR, so that's why I am confusing now.

**Comment by hyi** [comment]
@phuongdm I have done that, but run into issues with the following error:
```
ERROR: for data.local.org  Cannot start service data.local.org: driver failed programming external connectivity on endpoint data.local.org (ed678a69ca11b4fe8b091b4be200b531d9b0a03f4b13bc8b34c6e10fcda19f4b): Error starting userland proxy: listen tcp 0.0.0.0:1247: bind: An attempt was made to access a socket in a way forbidden by its access permissions.

ERROR: for data.local.org  Cannot start service data.local.org: driver failed programming external connectivity on endpoint data.local.org (ed678a69ca11b4fe8b091b4be200b531d9b0a03f4b13bc8b34c6e10fcda19f4b): Error starting userland proxy: listen tcp 0.0.0.0:1247: bind: An attempt was made to access a socket in a way forbidden by its access permissions.
Encountered errors while bringing up the project.
```
When I ran it yesterday for the first time, everything worked and I can ran tests. I was using my windows laptop, so when I was trying to bring all containers up this morning after restarting my laptop, I ran into the errors above. So I erased everything by running ```./local-dev-first-start-only.sh``` script to start over, but still run into the errors above. It just fails to bring up data.local.org container. I will try to manually erase everything to start over, but let me know what I should do to keep the running states the next day when I restart my laptop.

**Comment by phuongdm** [comment]
@hyi, 

Sometimes, I got unknown error from Docker Desktop on both Windows and Macbook, the fastest way to get rid of that is to restart laptop, then run those commands:

`docker rm -f $(docker ps -aq)`
`docker volume prune`
`docker system prune -a` 

I also pushed a small updated (set REMOTE_IRODS to be true on local_settings.template) to bypass the federated testing.

**Comment by hyi** [comment]
@phuongdm I finally got local dev env up and running on my windows laptop again after restarting my laptop and docker a couple of times and rebuilding. Really hope I don't have to do it again the next day after shutting down and restarting my laptop. I pulled your PR branch and can now reproduce the test error when iputting a file to user zone. I will look into it a bit when I have time, but need to fix those test failures for my PR first which is not related to federation.

**Comment by phuongdm** [comment]
@hyi,

I think that's permission problem, not federation problem. Probably, some default policies were changed from 4.1.8 to 4.2.6

File: hsctl [hunk 1]
```diff
@@ -273,8 +273,8 @@ migrate_hs() {
 
 manage_py_hs() {
     echo "MANAGEPY:"
-    echo "  - docker exec -u hydro-service -ti hydroshare python manage.py ${@:2}"
```
Comment (sblack-usu) [inline hsctl]: I'd prefer to leave -ti in... I use `./hsctl managepy shell` somewhat frequently.

Comment (phuongdm) [reply hsctl]: @sblack-usu, I absolutely want to keep it in `hsctl`. However, if we keep it, the script will not work on Windows in some cases.

**Comment by phuongdm** [comment]
@hyi, any update?

**Comment by hyi** [comment]
@phuongdm Sorry, I have not got time to look further into it yet, and since I am not that familiar with the local development setup environment, I am not sure how much help I can offer given that you have not figured out the permission issue. What is especially mysterious is that why the same setup works in Jenkins environment, but somehow does not work in local development environment. I'll try to find some time to look further into the issue during the next couple of weeks.

**Comment by hyi** [comment]
@phuongdm Yes, this works by design since when an iRODS user corresponding to the HS user is being created, HS iRODS proxy user is given owner privilege to the new iRODS user's home collection, hence the proxy user can put any file into the user's home collection/directory. See this code snippet for details: https://github.com/hydroshare/hydroshare/blob/develop/irods/create_user.sh#L15..L17. Can you check the new local development environment setup to make sure this iRODS permission is taken care of there similarly to the old iRODS local container setup with code snippet as shown above?

**Comment by phuongdm** [comment]
@hyi,

thanks, my new iRODS docker image has no those `.sh` files

**Comment by pkdash** [comment]
@phuongdm I am realizing now that I have created my current working branch for model program aggregation type from this branch (fix-test-error). Wondering if this PR will be merged soon.

**Comment by pkdash** [comment]
@hyi Is this ready for merge?

**Comment by hyi** [comment]
@pkdash I don't see @phuongdm has further commits since our last conversation here regarding allowing iRODS proxy user to upload files into personal iRODS user's user space in user zone, so that is still an issue remaining to be resolved. I guess this PR has improved the current local dev environment so I have no objection if you think this PR can be merged. But I think we should create an issue to capture the remaining problem which @phuongdm needs to further address when he gets a chance. You can use our last conversation as context for the new issue you create. I assume some user zone related tests will still fail in local development environment without further addressing the remaining permission issue.

**Comment by pkdash** [comment]
@hyi Since I created my working branch from this branch based on @phuongdm suggestion as I was not able to run unit tests locally, I am now having issues with updating my local branch with develop. I am hoping if this PR gets merged then updating my working branch with devlop should work. I am not really familiar with the work of this PR - so can't comment if this PR should be merged. However, I know work of this PR made running unit tests locally possible for me. 

@phuongdm See if you can create a separate issue to address @hyi 's concerns and merge this PR after @hyi approves it.

**Comment by cuahsimarko** [comment]
Spoke with Scott, we don't know what's going on here so someone will need to restart this work from first principles.

**Pull request closed without merging**